assets/js/components/Dashboard/NewPaMessage/StationsAndZones/SelectZonesPage.tsx

```diff
@@ -0,0 +1,541 @@
+    const signsWithoutSelectedRoute = fp.without(signIDs(allScreens), value);
+
+    const newScreens = groups.flatMap((group) =>
+      group.flatMap((zone) => signIDs(screensByZone[zone] ?? [])),
+    );
+
+    onChange(fp.uniq([...signsWithoutSelectedRoute, ...newScreens]));
+  };
+
+  const isAllSelected =
+    value.length > 0 &&
+    areAllGroupsSelected([LEFT_ZONES, MIDDLE_ZONES, RIGHT_ZONES]);
+  const isLeftSelected =
+    !isAllSelected && areAllGroupsSelected([LEFT_ZONES, MIDDLE_ZONES]);
+  const isRightSelected =
+    !isAllSelected && areAllGroupsSelected([MIDDLE_ZONES, RIGHT_ZONES]);
+
+  const hasRailFilter = () =>
+    [
+      "Green",
+      "Green-B",
+      "Green-C",
+      "Green-D",
+      "Green-E",
+      "Blue",
+      "Red",
+      "Orange",
+      "Mattapan",
+    ].includes(selectedRouteFilter);
+
+  const massSelectClassName = (selected: boolean) =>
+    selected ? "button-primary" : "button-secondary-outline";
+
+  const getFilterPillLabel = () => {
+    if (selectedRouteFilter.startsWith("Green-")) {
+      return `Green Line ${selectedRouteFilter.split("-")[1]}`;
+    } else {
+      return `${selectedRouteFilter} Line`;
+    }
+  };
+
+  return (
+    <div className="select-zones-page">
+      <div className="header">
+        <div className="title-and-edit">
+          <div className="title">Review Zones</div>
+          <div>
+            <Button
+              className="edit-button"
+              onClick={() => navigateTo(Page.STATIONS)}
+            >
+              Edit Stations
+            </Button>
+          </div>
+        </div>
+        <div className="buttons">
+          <Button
+            className="cancel-button"
+            onClick={() => navigateTo(Page.NEW)}
+          >
+            Cancel
+          </Button>
+          <Button
+            className="button-primary"
+            onClick={() => {
+              onSubmit(value);
+              navigateTo(Page.NEW);
+            }}
+          >
+            Done
+          </Button>
+        </div>
+      </div>
+      <div className="zone-selection">
+        <div className="filters-container">
+          <div>Service type</div>
+          <div className="filters">
+            {Object.keys(selectedRoutes).map((routeID) => {
+              let branchFilters;
+              if (routeID === "Green") {
+                branchFilters = selectedRoutes["Green"]
+                  .sort()
+                  .map((branchID) => {
+                    const branch = branchID.split("-")[1];
+
+                    return (
+                      <Button
+                        key={`branch-filter-${branch}`}
+                        onClick={() => setSelectedRouteFilter(branchID)}
+                        className={cx(
+                          "filter-button",
+                          selectedRouteFilter === branchID
+                            ? ROUTE_TO_CLASS_NAMES_MAP["Green"]
+                            : "filter-button-default",
```

Comment:
**quibble:** i would prefer to use the cascade here so that you could write something like this instead:

```javascript
className={cx('filter-button', {
  [[ROUTE_TO_CLASS_NAMES_MAP["Green"]]: selectedRouteFilter === branchID
})}
```